DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,113 @@
+# Developer guide
+
+This document describes how to set up a development environment for Calico, as well as how to build and test development code.
+
+## Requirements
+
+These build instructions assume you have a Linux build environment
+with:
+
+-  Docker 1.12+
+-  git
+-  make
+
+## Checking out the code
+
+Calico code is distributed across several different components, each of which lives in its own git repository. To build
+all of Calico, you will need to check out each repository into a common base directory (for example, `$GOPATH/src/github.com/projectcalico`).
+
+You will need the following repositories cloned on your development machine:
+
+```
+BASEDIR
+└───calico
+└───libcalico-go
+└───confd
+└───felix
+└───typha
+└───kube-controllers
+└───calicoctl
+└───app-policy
+└───pod2daemon
+└───node
+```
+
+## Building all the code
+
+Once the code is checked out, try building it to make sure your development environment is configured properly. Run the following command from
+within the `calico` repository to build container images from your locally checked out code.
+
+```
+make dev-image
+```
+
+This will build a number of `calico/X` images, tagged by git commit hash. To build images with a specific
+container registry, set the `REGISTRY` environment variable.
+
+```
+make dev-image REGISTRY=my-registry
```

Comment:
If I want to build locally and run it in a local cluster, can I simply make up a registry name for a non-existent registry, then use `dev-manifests ` to generate the configuration and deploy to my local cluster?

Reply:
Yep, exactly - no need to actually push the images.

I'd like to eventually add tooling for actually spinning up a local cluster (e.g., minikube or similar) so it becomes a one-click process. In the meantime, you can do exactly what you described.